DependencyInjection/Compiler/TwigExceptionPass.php

```diff
@@ -26,5 +26,9 @@ public function process(ContainerBuilder $container)
         if ($container->has('fos_rest.exception_listener') && $container->has('twig.exception_listener')) {
             $container->removeDefinition('twig.exception_listener');
         }
+
+        if (!$container->has('templating.engine.twig')) {
+            $container->removeDefinition('fos_rest.exception.twig_controller');
```

Comment:
This [optimization pass is always executed](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php#L59) so I think that we rely on it.